src/index.ts

```diff
@@ -58,11 +63,10 @@ export class PromiseQueue {
 				if (this.queue.length === 0) {
 					return;
 				}
-				let [timeoutSeconds] = process.hrtime();
-				timeoutSeconds += maxAgeSeconds;
 
 				const firstValid = this.queue.findIndex(
-					({ arrivalTime }) => arrivalTime[0] > timeoutSeconds,
+					({ arrivalTime }) =>
+						toSeconds(durationSince(arrivalTime)) > maxAgeSeconds,
```

Comment:
It would be better to keep the current behaviour where we we calculate the `timeoutSeconds` once and keep this function simple as it can potentially run multiple times

Reply:
Very good point. Fixed it!